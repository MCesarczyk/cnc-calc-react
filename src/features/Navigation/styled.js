import styled, { css } from "styled-components";
import { Link } from "react-router-dom";

export const NavigationWrapper = styled.div`
  align-self: flex-start;

  @media (max-width: ${({ theme }) => theme.breakpoint.md}) {
    display: none;
  }
`;

export const MobileNavWrapper = styled.div`
  display: none;

  @media (max-width: ${({ theme }) => theme.breakpoint.md}) {
    display: block;
  }
`;

export const List = styled.ul`
  padding: 0;
  list-style: none;
  margin-top: 6rem;
`;

export const Item = styled(Link)`
    color: ${({ theme }) => theme.color.primaryContrast};
    background-color: ${({ theme }) => theme.color.primary};
    display: block;
    font-size: 1.25rem;
    border-radius: 1.5rem;
    padding: 0.5rem 1.5rem;
    margin: 1rem;
    font-weight: 400;
    font-family: 'Ubuntu', sans-serif;
    margin: 0 auto 0.75rem auto;
    transition: all 0.3s ease-in-out;
    transition: transform 0.1s ease-in-out;

    &:hover {
      filter: brightness(1.5);
    }

    &:active {
      filter: brightness(1);
      transform: scale(0.98);
    }
    
    ${props => props.sidebar && css`
      border-radius: 0;
      border-bottom: 1px solid currentColor;
      background-color: transparent;
      transition: transform 0.1s ease-in-out;

      &:hover {
        filter: brightness(0.8);
        transform: scale(1.02);
      }

      &:active {
        filter: brightness(1);
        transform: scale(0.98);
      }
    `}
`;