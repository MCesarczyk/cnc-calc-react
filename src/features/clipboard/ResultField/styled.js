import styled, { css } from "styled-components";

export const ResultWrapper = styled.div`
  display: flex;
  flex-basis: 16rem;

  @media (max-width: ${({ theme }) => theme.breakpoint.xs}) {
    flex-basis: unset;
    width: 100%;
  }
`

export const Result = styled.div`
    min-width: 16rem;
    border: solid 1px ${({ theme }) => theme.color.primary};
    border-radius: 5px;
    padding: 5px;
    color: ${({ theme }) => theme.color.primaryText};

    ${({ placeholder }) => placeholder && css`
      color: ${({ theme }) => theme.color.placeholder};
    `}
`;

export const ResultCopyButton = styled.button`
  background: transparent;
  border: none;
  outline: none;
  color: ${({ theme }) => theme.color.copyButtonFont};
  margin-left: 0.5rem;

  &:hover {
    filter: brightness(1.5);
  }

  &:active {
    filter: brightness(0.5);
  }
`