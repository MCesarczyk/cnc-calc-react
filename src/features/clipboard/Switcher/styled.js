import styled from "styled-components";

export const SwitcherWrapper = styled.div`
  margin-top: 1.5rem;
  padding-left: 2.5rem;
`

export const SwitcherLabel = styled.label`
  font-variant: small-caps;
  font-weight: 700;
  margin-right: 1rem;
`

export const Switcher = styled.button``

export const SwitcherIndicator = styled.div``