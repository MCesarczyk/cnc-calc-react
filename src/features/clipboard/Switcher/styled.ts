import styled, { css } from 'styled-components';

export const SwitcherWrapper = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  margin-top: 1.5rem;
`;

export const SwitcherLabel = styled.label`
  font-variant: small-caps;
  font-weight: 700;
  margin-right: 1rem;
  cursor: pointer;
`;

export const Switcher = styled.button`
  width: 3rem;
  height: 2rem;
  background-color: #d9d9d9;
  border: 2px solid #525252;
  border-radius: 1rem;
`;

export const SwitcherIndicator = styled.div<{ $memorymode?: boolean }>`
  width: 2rem;
  height: 2rem;
  border: 2px solid #525252;
  border-radius: 1rem;
  background-color: #d9d9d9;
  position: relative;
  left: -0.5rem;
  top: -3px;
  transition: all 0.3s ease-in-out;

  ${({ $memorymode }) =>
    $memorymode &&
    css`
      transform: translateX(1.2rem);
      background-color: #1e3246;
    `}
`;
