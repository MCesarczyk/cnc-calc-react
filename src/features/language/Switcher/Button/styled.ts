import styled, { css } from 'styled-components';

export const SwitcherButton = styled.button<{ $active?: string }>`
  color: ${({ theme }) => theme.color.primary};
  background-color: ${({ theme }) => theme.color.primaryContrast};
  background-repeat: no-repeat;
  background-position: center;
  background-size: cover;
  border: 2px solid ${({ theme }) => theme.color.primary};
  border-radius: 2px;

  ${({ $active }) =>
    $active === 'active' &&
    css`
      border-color: ${({ theme }) => theme.color.primaryContrast};
    `}

  &:hover {
    filter: brightness(125%);
  }

  &:active {
    filter: brightness(150%);
  }
`;
