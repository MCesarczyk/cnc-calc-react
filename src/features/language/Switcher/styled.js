import styled from "styled-components";

export const SwitcherWrapper = styled.div`
    display: flex;
    
    @media (max-width: ${({ theme }) => theme.breakpoint.sm}) {
        flex-direction: column;
    }
`;