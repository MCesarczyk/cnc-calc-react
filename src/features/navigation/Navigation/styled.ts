import styled from 'styled-components';

export const NavigationWrapper = styled.div`
  align-self: flex-start;
  margin-top: 2rem;

  @media (max-width: ${({ theme }) => theme.breakpoint.md}) {
    display: none;
  }
`;

export const MobileNavWrapper = styled.div`
  display: none;

  @media (max-width: ${({ theme }) => theme.breakpoint.md}) {
    display: block;
  }
`;
