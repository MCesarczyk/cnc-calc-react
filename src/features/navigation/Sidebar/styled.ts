import styled from 'styled-components';

export const SidebarActivationArea = styled.div`
  position: absolute;
  left: 0;
  top: 0;
  width: 10px;
  height: 100%;
  background: transparent;
`;

export const SidebarNavigation = styled.div<{ $visible?: boolean }>`
  position: absolute;
  left: 0;
  top: 0;
  height: 100%;
  color: ${({ theme }) => theme.color.primaryContrast};
  background-color: ${({ theme }) => theme.color.primary};
  padding: 1rem;
  padding-top: 6rem;
  display: none;

  @media (max-width: ${({ theme }) => theme.breakpoint.md}) {
    display: ${({ $visible }) => $visible && 'block'};
  }
`;
