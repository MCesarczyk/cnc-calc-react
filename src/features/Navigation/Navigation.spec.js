import React from "react";
import renderer from "react-test-renderer";
import { ThemeProvider } from "styled-components";
import { theme } from "../../theme";
import "jest-styled-components";
import { Item, List } from "./styled";
import { BrowserRouter } from "react-router-dom";

test("Navigation list rendered properly", () => {
  const listComponent = renderer.create(
    <ThemeProvider theme={theme} >
      <List />
    </ThemeProvider>
  );

  let list = listComponent.toJSON();
  expect(list).toMatchSnapshot();
});

test("List item rendered properly", () => {
  const listItemComponent = renderer.create(
    <ThemeProvider theme={theme} >
      <BrowserRouter>
        <Item to="#" />
      </BrowserRouter>
    </ThemeProvider>
  );

  let item = listItemComponent.toJSON();
  expect(item).toMatchSnapshot();
});