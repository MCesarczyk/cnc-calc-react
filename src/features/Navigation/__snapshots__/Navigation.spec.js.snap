// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List item rendered properly 1`] = `
.c0 {
  color: #fff;
  background-color: #1e3246;
  display: block;
  font-size: 1.25rem;
  border-radius: 1.5rem;
  padding: 0.5rem 1.5rem;
  margin: 1rem;
  font-weight: 400;
  font-family: 'Ubuntu',sans-serif;
  margin: 0 auto 0.75rem auto;
  -webkit-transition: all 0.3s ease-in-out;
  transition: all 0.3s ease-in-out;
  -webkit-transition: -webkit-transform 0.1s ease-in-out;
  -webkit-transition: transform 0.1s ease-in-out;
  transition: transform 0.1s ease-in-out;
}

.c0:hover {
  -webkit-filter: brightness(1.5);
  filter: brightness(1.5);
}

.c0:active {
  -webkit-filter: brightness(1);
  filter: brightness(1);
  -webkit-transform: scale(0.98);
  -ms-transform: scale(0.98);
  transform: scale(0.98);
}

<a
  className="c0"
  href="/"
  onClick={[Function]}
/>
`;

exports[`Navigation list rendered properly 1`] = `
.c0 {
  padding: 0;
  list-style: none;
  margin-top: 6rem;
}

<ul
  className="c0"
/>
`;
